docs/development/design/exit-node.md

```diff
@@ -0,0 +1,145 @@
+x.227.102.183:44728     6/CwH/gzz9jdKnxeVpWFivycFQMBkniLeBNFQq0+f04=
+
+$ wget -qO - icanhazip.com
+x.227.102.183
+
+$ sudo nexctl nexd exit-node disable
+Successfully disabled exit node client on this device
+
+# After subscribing to the exit-node, the client is now originating traffic from the exit node.
+$ wget -qO - icanhazip.com
+x.31.152.254
+```
+
+A diagram of the traffic flow is as follows:
+
+```mermaid
+graph LR
+subgraph Nexodus Mesh with All Devices Peered to one another
+Nex_Device_4["Nexd-1 (100.100.0.1)
+exit-node client"]
+Nex_Device_5["Nexd-2 (100.100.0.2)
+exit-node client"]
+Nex_Device_6["Nexd-3 (100.100.0.3)
+exit-node client"]
+Nexodus_Network_Router_B["Nexd-4 Exit Node 
+wg0 100.100.0.4
+eth(n) 172.16.100.10/24"]
+end
+
+subgraph External Network 
+Internet["Network or Internet
+Receives Traffic from
+the Nexodus Network
+as sourced from eth(n)
+172.16.100.10 via SNAT"]
+end
+
+ Nexodus_Network_Router_B-- "eth(n)" --> Internet
+Nex_Device_4 -- wg0 --> Nexodus_Network_Router_B
+Nex_Device_5 -- wg0 --> Nexodus_Network_Router_B
+Nex_Device_6 -- wg0 --> Nexodus_Network_Router_B
+```
+
+### Client Out of Band Services
+
+- Since all traffic is being tunneled there is a high likelihood that at some point the device will lose connectivity to the exit node and become orphaned. Because of that, we are baking in as many safeguards as possible and options to recover connectivity.
+
+1. Nexodus services will travel out of band via the device's interface containing a default route via Linux policy routing. A new routing table is created with a default route pointing to the same default route in the default namespace. Then using combinations of nftables and rules in the Linux RPDB (routing policy database) traffic destined for the OOB services is mangled and forwarded bypassing the Nexodus driver interface (for example wg0). The initial out of band services are:
+   - HTTPS port 443 to the API server's destination IP(s). If an exit node becomes unavailable, the exit node client device would no longer be able to reach the device. This would orphan the node and require a user to run nexctl to disable the exit node subscription. To allow an administrator or user of many devices to be able to dis-enroll a device from the web UI, we need to maintain api-server connectivity even if an exit node is unavailable.
+   - DNS UDP 53. This should like be discovered via host discovery. There could potentially be scenarios where a DNS is only available via the Nexd mesh but that would be fairly atypical and should be tackled with Nexodus DNS implementation. This rule could also be narrowed if there was a host entry added to the device the agent is running on.
+   - STUN requests, UDP 3478 and 19302 for the RFC defined and Google STUN ports. This is required because once a device is forwarding all traffic through the exit node, the reflexive address for the client device would take on the same reflexive address of the exit node.
```

Comment:
I wrote the comment below before I realized we had this discussion in a previous comment thread. It would be great if you could update the doc to include the reasoning why you think this type of traffic should be handled correctly.

----

I don't think this is quite right. This will break any other application that uses STUN.

I think we need to find a way to only send our own stun requests outside of the mesh. Any other application using STUN should go through the exit node.

Reply:
Really the only way to do it with PBR is run our own STUN that no other applications would use. Maybe marking traffic somehow but I would need to circle back on that. One thing to limit the issue would be only peel off `19302` the custom Google port since that is what we are currently using and leave the RFC port 3478 alone in the short term.

Reply:
I'm OK just documenting this as a caveat of using exit nodes for now